### PR TITLE
add float-cell cases

### PR DESCRIPTION
**Pull request by arnej27959**
* also, make input values exactly representable as floats

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe please review and merge

### CODE REVIEW

File: eval/src/apps/tensor_conformance/generate.cpp [hunk 1]
```diff
@@ -131,43 +146,74 @@ void generate_op2_join(const vespalib::string &op2_expr, const Sequence &seq, Te
...
 
 void generate_xw_product(TestBuilder &dst) {
     auto matrix = spec({x(2),y(3)}, Seq({ 3, 5, 7, 11, 13, 17 }));
+    auto fmatrix = spec(float_cells({x(2),y(3)}), Seq({ 3, 5, 7, 11, 13, 17 }));
     dst.add("reduce(a*b,sum,x)", {{"a", spec(x(2), Seq({ 1, 2 }))}, {"b", matrix}},
             spec(y(3), Seq({(1*3+2*11),(1*5+2*13),(1*7+2*17)})));
+    dst.add("reduce(a*b,sum,x)",
+            {{"a", spec(float_cells({x(2)}), Seq({ 1, 2 }))}, {"b", matrix}},
+            spec(y(3), Seq({(1*3+2*11),(1*5+2*13),(1*7+2*17)})));
+    dst.add("reduce(a*b,sum,x)",
```
Comment (havardpe) [inline eval/src/apps/tensor_conformance/generate.cpp]: consider adding test for float matrix and double vector

File: eval/src/apps/tensor_conformance/generate.cpp [hunk 2]
```diff
@@ -177,6 +223,9 @@ void generate_xw_product(TestBuilder &dst) {
 void generate_tensor_concat(TestBuilder &dst) {
     dst.add("concat(a,b,x)", {{"a", spec(10.0)}, {"b", spec(20.0)}}, spec(x(2), Seq({10.0, 20.0})));
     dst.add("concat(a,b,x)", {{"a", spec(x(1), Seq({10.0}))}, {"b", spec(20.0)}}, spec(x(2), Seq({10.0, 20.0})));
+    dst.add("concat(a,b,x)", {{"a", spec(float_cells({x(1)}), Seq({10.0}))},
```
Comment (havardpe) [inline eval/src/apps/tensor_conformance/generate.cpp]: is this a duplicate?

**Review by havardpe — COMMENTED** [review]
only a couple of minor comments

File: eval/src/apps/tensor_conformance/generate.cpp (hunk 1 above)
Comment (arnej27959) [reply eval/src/apps/tensor_conformance/generate.cpp]: added

File: eval/src/apps/tensor_conformance/generate.cpp (hunk 2 above)
Comment (arnej27959) [reply eval/src/apps/tensor_conformance/generate.cpp]: removed duplicate

**Pull request merged**